lib/proxy.js

```diff
@@ -174,7 +174,7 @@ function ReverseProxy(opts){
     if(target){
       proxy.ws(req, socket, head, {target: target});
     }else{
-      notFound(socket);
+      respondNotFound(req, socket);
```

Comment:
I'm not sure if it would be valid anyway to attempt to send a "not found" over a socket.
Actually, wouldn't the destination already have had to be found before the server can emit the "upgrade" event? So there would be no need for the custom error message because that should never happen in practice (unless you have the proxy settings changing in the middle of a connection or something weird like that)